src/modules/ekf2/EKF/covariance.cpp

```diff
@@ -103,14 +103,43 @@ void Ekf::initialiseCovariance()
 
 void Ekf::predictCovariance(const imuSample &imu_delayed)
 {
-	// Use average update interval to reduce accumulated covariance prediction errors due to small single frame dt values
-	const float dt = _dt_ekf_avg;
+	// gyro_bias: clear covariances if inhibited
+	for (unsigned index = 0; index < State::gyro_bias.dof; index++) {
+		if (_gyro_bias_inhibit[index]) {
+			P.uncorrelateCovariance<1>(State::gyro_bias.idx + index);
+		}
+	}
+
+	// accel_bias: clear covariances if inhibited
+	for (unsigned index = 0; index < State::accel_bias.dof; index++) {
+		if (_accel_bias_inhibit[index]) {
+			P.uncorrelateCovariance<1>(State::accel_bias.idx + index);
+		}
+	}
+
+#if defined(CONFIG_EKF2_MAGNETOMETER)
+	// mag_I, mag_B: clear covariances if inactive
+	if (!_control_status.flags.mag) {
+		P.uncorrelateCovarianceBlock<State::mag_I.dof>(State::mag_I.idx);
+		P.uncorrelateCovarianceBlock<State::mag_B.dof>(State::mag_B.idx);
+	}
+#endif // CONFIG_EKF2_MAGNETOMETER
+
+#if defined(CONFIG_EKF2_WIND)
+	// wind_vel: clear covariances if inactive
+	if (!_control_status.flags.wind) {
+		P.uncorrelateCovarianceBlock<State::wind_vel.dof>(State::wind_vel.idx);
```

Comment:
Part of wind stop logic instead?

Reply:
Since the covariance is reset when stopping wind (https://github.com/PX4/PX4-Autopilot/pull/22597/files#diff-3da6892507fab6d658d0f3849fcfda126ae07e2305ea89c9a58ebee6bbe843d6R63) we can just remove it. (I'll add a commit)